### DISCUSSION TITLE
객체지향의 사실과 오해 book talk

### DISCUSSION THREAD

**Discussion by jongfeel** [ThinkAboutSoftware/OnlineSelfCodingGroup#27] [Book talk]
@hi-hyein 님이 [객체지향의 사실과 오해](http://aladin.kr/p/aGexE)라는 책을 읽고 정리하기 시작했는데
저는 5년 전 책을 읽었고 그 기억만으로 얘기하기에는 조금 부족한 감이 있어서
저도 책을 다시 읽고 book talk를 해보면 어떻겠냐고 혜인님에게도 제안하였고
역시 discussions 탭에 만들게 되었습니다.

논의할 범위와 주제를 선정한 후에, 언제 얘기할지 오프라인이 좋을지 온라인이 좋을지도 정해보면 좋겠습니다.

#24 #25 에 @yoonucho 님과 이미 오브젝트 주제 관련해서 한번 진행한 적이 있습니다.
참고해 보시면 좋습니다.

추가)
pull request 만들 때 어떤 폴더 이름으로 할지 고민되실 까봐 책 영어 제목을 미리 적어 둡니다.
booktalk/TheEssenceOfObjectOrientation

**Comment by hi-hyein** [comment]
@jongfeel 

드디어 ...그 날이 왔군요..!
일단 책은 금주안에 완독할 것 같습니다 😄 
범위와 주제는 좀 더 생각해 볼게요. 북토크는 난생 처음이라 감이 안잡히네요!
혹시 멘토님이 좋다고 생각하는 범위와 주제가 있다면 말씀해 주세요! 

+++ 시간은 3월 8일 이후로, 온라인으로 진행하는 게 좋을 것 같아요!

**Reply by jongfeel** [reply]
저도 책은 절반 정도 읽어 나가고 있습니다.

북토크가 다른 걸 알려드리면

- book review: 책에 있는 내용을 왜곡(?) 없이 정리해서 얘기하고 자신의 소감, 감상을 추가함, 리뷰를 보는 사람의 의견은 의견일 뿐 논의 대상은 아닐 가능성이 높음
- book talk: 책에 있는 내용에 대해 자신의 생각을 담아 정리하고, 질문 거리를 만들어 서로의 생각과 의견을 얘기할 수 있어야 함. 서로 의견을 주고 받는 과정이 중요함.

이렇게 보시면 됩니다.

범위는 책 전체 말고 여러번 나눠서 하면 좋을 것 같습니다. 2~3 챕터를 묶으면 세번 정도면 할 수 있을 거 같아요.

3월 10일 수요일이 자리가 나니까 일단 그 날로 잡아 보도록 하겠습니다.

못 정하겠으면 챕터3 까지 해볼까요?

**Reply by hi-hyein** [reply]
네?절반이요? 후덜덜ㅋㅋㅋㅋㅋㅋㅋㅋㅋ
설명 감사합니다! 윤우님 북토크 내용과 같이 보니 잘 이해되네요!
일정은 3월 10일 수요일 좋습니다!
시간은 오후6시 이후는 모두 가능하니 멘토님도 가능한 시간에 맞추면 될 것 같습니다
챕터3까지 진행하는 것도 좋아요! 짝짝짝짝
범위 정하는걸 어렵게 생각했었네요ㅠㅠ
그럼 한 번에 한 챕터씩 진행하는 것으로 알고 준비하겠습니다!

**Reply by jongfeel** [reply]
아 오해가 있게 설명을 드린 것 같네요

> 2 ~ 3 챕터를 묶으면 세번 정도면 할 수 있을 거 같아요.

이게 저희가 정한 3 챕터를 세번 해서 한 챕터에 한번씩 한다는 뜻이 아니고요

한번에 3 챕터를 진행해서 세번 진행했을 때 9 챕터가 진행 된다는 뜻입니다.

이러면 이해가 될까요?

그래서 한번에 챕터 1, 2, 3을 진행하고 싶은 건데 분량이나 기간 조절이 필요하시면 다시 얘기해 주시면 됩니다.

**Reply by hi-hyein** [reply]
아하! 넵 이해됐어요 분량은 충분한 것 같습니다
3챕터씩 진행하는 것으로~~~~ 👍👍

**Reply by jongfeel** [reply]
네 그러면 챕터 하나에 pull request 하나씩 만들어 보면 어떨까요?

branch name is

- booktalk/TheEssenceOfObjectOrientation/chapter01
- booktalk/TheEssenceOfObjectOrientation/chapter02
- booktalk/TheEssenceOfObjectOrientation/chapter03

이렇게 되겠죠.

그리고 3월 10일 수요일 리뷰니까
3월 7일 일요일 까지 챕터 3 까지 pull reuqest 3개가 요청되야 합니다. 충분한 리뷰를 할 수 있는 시간이 필요하니까요.
물론 그 전에 올려주셔서 리뷰를 진행하면 더 좋습니다.

**Reply by hi-hyein** [reply]
ㅎㅎ네 좋습니다!

**Comment by jongfeel** [comment]
다음 일정입니다.
- chapter 04, chapter 05 동일한 방식으로 각각 pull request 리뷰 진행
- 온라인 리뷰 미팅: 3월 24일 수요일

**Reply by jongfeel** [reply]
원활한 온라인 리뷰 진행을 위해 3월 21일 일요일까지 일단 pull request를 올려주셔야 합니다! 😄

**Comment by hi-hyein** [comment]
@jongfeel 
멘토님 4월 23일 금요일 일정이 생겨 리뷰진행이 여려울 것 같아요!
멘토님 시간도 괜찮으시면 바로 다음주 4월 26일 ~ 30일에 진행했으면 하는데 어떠신가요!
가능하시면 편하신 날에 초대 부탁드립니다 🙏 🙏 🙏 🙏

**Reply by jongfeel** [reply]
다음주는 일정이 다 비네요 26일로 바로 갈까요?

내용이 오래 되서 26일 전에 한번 내용 읽고 오는게 좋을 것 같습니다.
저도 그래야 할 것 같고요.

**Reply by hi-hyein** [reply]
오 다행이에요! 26일 완전좋아요!!
내용도 한번 다시 정리해보고 오겠습니다~~
